File: VB/Scaffolding.ManyToMany/SchoolContextDataModel/SchoolContextMetadata.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports Scaffolding.ManyToMany.Localization
Imports Scaffolding.ManyToMany.Model

Namespace Scaffolding.ManyToMany.SchoolContextDataModel

	Public Class SchoolContextMetadataProvider
		Public Shared Sub BuildMetadata(ByVal builder As MetadataBuilder(Of Course))
			builder.DisplayName(SchoolContextResources.Course)
			builder.Property(Function(x) x.CourseId).DisplayName(SchoolContextResources.Course_CourseId)
			builder.Property(Function(x) x.Title).DisplayName(SchoolContextResources.Course_Title)
		End Sub
		Public Shared Sub BuildMetadata(ByVal builder As MetadataBuilder(Of Student))
			builder.DisplayName(SchoolContextResources.Student)
			builder.Property(Function(x) x.StudentId).DisplayName(SchoolContextResources.Student_StudentId)
			builder.Property(Function(x) x.FirstName).DisplayName(SchoolContextResources.Student_FirstName)
			builder.Property(Function(x) x.LastName).DisplayName(SchoolContextResources.Student_LastName)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.ManyToMany/SchoolContextDataModel/SchoolContextMetadata.cs
using DevExpress.Mvvm.DataAnnotations;
using Scaffolding.ManyToMany.Localization;
using Scaffolding.ManyToMany.Model;

namespace Scaffolding.ManyToMany.SchoolContextDataModel {

    public class SchoolContextMetadataProvider {
        public static void BuildMetadata(MetadataBuilder<Course> builder) {
            builder.DisplayName(SchoolContextResources.Course);
            builder.Property(x => x.CourseId).DisplayName(SchoolContextResources.Course_CourseId);
            builder.Property(x => x.Title).DisplayName(SchoolContextResources.Course_Title);
        }
        public static void BuildMetadata(MetadataBuilder<Student> builder) {
            builder.DisplayName(SchoolContextResources.Student);
            builder.Property(x => x.StudentId).DisplayName(SchoolContextResources.Student_StudentId);
            builder.Property(x => x.FirstName).DisplayName(SchoolContextResources.Student_FirstName);
            builder.Property(x => x.LastName).DisplayName(SchoolContextResources.Student_LastName);
        }
    }
}